.github/workflows/build-and-test-toolchain.yml

```diff
@@ -64,15 +64,20 @@ env:
   PLATFORM: ${{ inputs.platform || 'w64-mingw32' }}
   CRT: ${{ inputs.crt || 'msvcrt' }}
 
-  SOURCE_PATH: ~/code
-  DOWNLOADS_PATH: ~/downloads
-  ARTIFACT_PATH: ~/artifacts
+  SOURCE_PATH: /root/code
```

Comment:
`~` expansion does not work in this case.

Reply:
what about using something like ./code?

Reply:
I think it would be quite dangerous. What would be the benefit of using the relative paths? I suppose it might change what is the $GITHUB_WORKSPACE mapped to inside of the WSL environment. Note that this path needs to be specified for a smoot transition of paths between the Windows host and WSL environment.

Reply:
I understand, the idea is to avoid mapping to particular user.  It is ok for me, however I would move execution to a  script where the user folder can be detected and specified.

Reply:
I understand the concern about the user name mapping. Though I don't understand what you mean by moving the detection of the user to the script? The WSL environment GHA action runs as `root` in the WSL container by default and I don't think the host is able to detect it from outside. Do you suggest that the scripts inside the WSL container would output the directories mapping to the GHA step output and the host would use a variable from that step output?

Reply:
Something line "source .github/scripts/config.sh" before cloning the repos.

Reply:
Do I understand correctly that you are suggesting that the host would alter the `.github/scripts/config.sh` to contain absolute paths before setting up the WSL environment?

Reply:
Yes, this is an idea for this case.

Reply:
Then I don't understand what is the difference/benefit? Altering the `config.sh` via file modification seems more cumbersome that just setting the environment variables and leveraging `VAR=${VAR:-default}` Bash syntax.

What I would do to bullet-proof this setting is to either force the specific WSL user ()https://github.com/marketplace/actions/setup-wsl#wsl-shell-user:
```
- uses: Vampire/setup-wsl@v3
  with:
      wsl-shell-user: root
``` 
Or by using other base folder than `root` home folder, e.g.:
```
  SOURCE_PATH: /tmp/code
  DOWNLOADS_PATH: /tmp/downloads
  ARTIFACT_PATH: /tmp/artifacts
```

Which one would you prefer?

Reply:
I took a look again in config.sh
SOURCE_PATH=${SOURCE_PATH:-$PWD/code}

Based on the current workflow, ./code looks acceptable for me.

Reply:
This does not work, moving to draf for now...

Reply:
Issues fixed, ready for review.

Reply:
Thanks, it looks better for me now.